### ISSUE TITLE
Improper cursor position for multichar unicode

### ISSUE THREAD

**Issue by Owen000** [nvimdev/dashboard-nvim#465]
**Describe the bug**
I am still experiencing the issue that was solved here:
https://github.com/nvimdev/dashboard-nvim/issues/422
multibyte chars move my cursor to the improper positioning.

**To Reproduce**
1. ```touch repro.lua```
2. paste in

```
local root = vim.fn.fnamemodify("./repro", ":p")
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

local plugins = {
  {
    "nvimdev/dashboard-nvim",
    event = "VimEnter",
    opts = function()
      local opts = {
        theme = "doom",
        config = {
          header = {
            "                                                       ",
            "                                                       ",
            "                                                       ",
            " ███╗   ██╗ ███████╗ ██████╗  ██╗   ██╗ ██╗ ███╗   ███╗",
            " ████╗  ██║ ██╔════╝██╔═══██╗ ██║   ██║ ██║ ████╗ ████║",
            " ██╔██╗ ██║ █████╗  ██║   ██║ ██║   ██║ ██║ ██╔████╔██║",
            " ██║╚██╗██║ ██╔══╝  ██║   ██║ ╚██╗ ██╔╝ ██║ ██║╚██╔╝██║",
            " ██║ ╚████║ ███████╗╚██████╔╝  ╚████╔╝  ██║ ██║ ╚═╝ ██║",
            " ╚═╝  ╚═══╝ ╚══════╝ ╚═════╝    ╚═══╝   ╚═╝ ╚═╝     ╚═╝",
            "                                                       ",
            "                                                       ",
            "                                                       ",
            "                                                       ",
          },
          center = {
            {
              icon = "󰈞  ",
              desc = "Find File                              ",
              action = function()
                require("telescope.builtin").find_files()
              end,
              key = "<Leader> p f",
            },
            {
              icon = "󰈢  ",
              desc = "Recently opened files                   ",
              action = function()
                require("telescope.builtin").oldfiles()
              end,
              key = "<Leader> p o",
            },
            {
              icon = "  ",
              desc = "Open Nvim config                        ",
              action = function()
                require("telescope.builtin").find_files({ cwd = vim.fn.expand("~/.config/nvim") })
              end,
              key = "<Leader> e c",
            },
            {
              icon = "  ",
              desc = "New file                                ",
              action = "enew",
              key = "e",
            },
            {
              icon = "󰗼  ",
              desc = "Quit Nvim                               ",
              action = "qa",
              key = "q",
            },
          },
        },
      }
      return opts
    end,
  },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})


```
3. nvim -u repro.lua
4. run :Dashboard and hit j 2 times, you will see the issue on the "gear" icon and the "one file" icon.

**Expected behavior**
Not moving my cursor further

**Screenshots**
Undesired behavior:
![image](https://github.com/nvimdev/dashboard-nvim/assets/67721175/0efe0e4a-6bb1-4792-8786-fb2c8cf11d56)

Desired behavior
![image](https://github.com/nvimdev/dashboard-nvim/assets/67721175/c11744f5-0f1c-4cc7-b6b5-e2ad3d95f531)
(being on the first letter)

**Comment by Owen000** [comment]
Interesting note, I found that if I set the very topmost icon (Find File) to  (the gear icon that had the issue) it doesn't break.

**Comment by polirritmico** [comment]
I could reproduce it

**Comment by polirritmico** [comment]
Fixed by #470
